web/src/features/hfiCalculator/FormatTableAsCSV.ts

```diff
@@ -0,0 +1,253 @@
+        `,, ${dailyIntensityGroup}, ${fireStarts}, ${areaDailyPrepLevel},`
+      )
+    })
+    areaWeeklySummaryString = areaWeeklySummaryString.concat(
+      `${highestMeanIntensityGroup}, ${calcPrepLevel}`
+    )
+    return areaWeeklySummaryString
+  }
+
+  public static exportWeeklyRowsAsStrings = (
+    fireCentres: Record<string, FireCentre>,
+    dailies: StationDaily[]
+  ): string => {
+    const rowsAsStrings: string[] = []
+    const dateSet = new Set()
+
+    dailies.flatMap(dailyRecord => {
+      dateSet.add(
+        `${dailyRecord.date.weekdayShort} ${dailyRecord.date.monthShort} ${dailyRecord.date.day}`
+      )
+    })
+    // build header row of dates
+    rowsAsStrings.push(
+      Array(NUM_STATION_DATA_COLS).join(',').concat(Array.from(dateSet).join(',,,,,'))
+    )
+    rowsAsStrings.push(weeklyTableColumnLabels.toString())
+
+    Object.entries(fireCentres).forEach(([, centre]) => {
+      rowsAsStrings.push(centre.name)
+      Object.entries(centre.planning_areas)
+        .sort((a, b) => (a[1].name.slice(-3) < b[1].name.slice(-3) ? -1 : 1)) // sort by zone code
+        .forEach(([, area]) => {
+          rowsAsStrings.push(this.buildAreaWeeklySummaryString(area, dailies))
+          Object.entries(area.stations).forEach(([, station]) => {
+            const dailiesForStation = getDailiesByStationCode(dailies, station.code)
+            const grassCureError = !isValidGrassCure(dailies[0], station.station_props)
+
+            const rowArray: string[] = []
+
+            rowArray.push(station.station_props.name + ' (' + station.code + ')')
+            if (
+              isUndefined(station.station_props.elevation) ||
+              isNull(station.station_props.elevation)
+            ) {
+              rowArray.push('ND')
+            } else {
+              rowArray.push(station.station_props.elevation.toString())
+            }
+            rowArray.push(station.station_props.fuel_type.abbrev)
+            rowArray.push(
+              grassCureError ? 'ERROR' : printGrassCurePercentage(dailiesForStation[0])
+            )
+
+            dailiesForStation.forEach(day => {
+              rowArray.push(
+                !isUndefined(day) && !grassCureError
+                  ? day.rate_of_spread.toFixed(DECIMAL_PLACES)
+                  : 'ND'
+              )
+              rowArray.push(
+                !isUndefined(day) && !grassCureError
+                  ? day.hfi.toFixed(DECIMAL_PLACES)
+                  : 'ND'
+              )
+              rowArray.push(
+                !isUndefined(day) && !grassCureError
+                  ? day.intensity_group.toString()
+                  : 'ND'
+              )
+              rowArray.push(Array(NUM_WEEKLY_SUMMARY_CELLS).join(','))
+            })
+
+            rowArray.push(Array(NUM_WEEKLY_SUMMARY_CELLS).join(','))
+
+            rowsAsStrings.push(CSV.stringify(rowArray))
+          })
+        })
+    })
+
+    return rowsAsStrings.join('\r')
```

Comment:
Just curious, what does this character do and is it platform agnostic?

Reply:
`\r` is carriage return. Turns out that it's no longer necessary with the csv-string Typescript module we're using